lib/ansible/module_utils/basic.py

```diff
@@ -1483,31 +1483,28 @@ def _check_type_dict(self, value):
         raise TypeError('%s cannot be converted to a dict' % type(value))
 
     def _check_type_bool(self, value):
-        if isinstance(value, bool):
-            return value
-
-        if isinstance(value, string_types) or isinstance(value, int):
-            return self.boolean(value)
-
-        raise TypeError('%s cannot be converted to a bool' % type(value))
+        if not isinstance(value, bool):
+            try:
+                return self.boolean(value)
```

Comment:
Would love to also modify self.boolean to raise a TypeError but unfortunately that's a public method that modules sometimes use :-(  I guess we can live with that doing something different and making a note to change it when we refactor basic.py.